### PR TITLE
GEOMESA-2696 Converters - pass nulls through type inference

### PR DESCRIPTION
**Pull request by elahrvivaz**

### CODE REVIEW

File: geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/transforms/GeometryFunctionFactory.scala [hunk 1]
```diff
@@ -34,55 +34,63 @@ class GeometryFunctionFactory extends TransformerFunctionFactory {
     case Array(g: Array[Byte]) => WKBUtils.read(g).asInstanceOf[Point]
     case Array(x: Float, y: Double) => gf.createPoint(new Coordinate(x, y))
     case Array(x: Double, y: Float) => gf.createPoint(new Coordinate(x, y))
+    case Array(null) | Array(null, null) => null
```
Comment (jnh5y) [inline geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/transforms/GeometryFunctionFactory.scala:37]: Idle note:  Looks like this doesn't support XYZ, XYM, XYZM, right?

Comment (elahrvivaz) [reply geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/transforms/GeometryFunctionFactory.scala:37]: right, we should probably add that

Comment (jnh5y) [reply geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/transforms/GeometryFunctionFactory.scala:37]: Are you knocking that out in this PR or another ticket?

Comment (elahrvivaz) [reply geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/transforms/GeometryFunctionFactory.scala:37]: i'll open a ticket

**Pull request merged**